docs/source/guide/migrationec2.rst

```diff
@@ -0,0 +1,123 @@
+.. _guide_migration_ec2:
+
+Amazon EC2
+==========
+Boto 2.x contains a number of customizations to make working with Amazon EC2 instances, storage and networks easy.
+
+Creating the Connection
+-----------------------
+Boto 3 has both low-level clients and higher-level resources. For Amazon EC2, the higher-level resources are the most similar to Boto 2.x's ``ec2`` and ``vpc`` modules::
+
+    # Boto 2.x
+    import boto
+    ec2_connection = boto.connect_ec2()
+    vpc_connection = boto.connect_vpc()
+
+    # Boto 3
+    import boto3
+    ec2 = boto3.resource('ec2')
+
+Launching New Instances
+-----------------------
+Launching new instances requires an image ID and the number of instances to launch. It can also take several optional parameters, such as the instance type and security group::
+
+    # Boto 2.x
+    ec2_connection.run_instances('<ami-image-id>')
+
+    # Boto 3
+    ec2.create_instances(ImageId='<ami-image-id>', MinCount=1, MaxCount=5)
+
+Stopping & Terminating Instances
+--------------------------------
+Stopping and terminating multiple instances given a list of instance IDs uses Boto 3 collection filtering::
+
+    ids = ['instance-id-1', 'instance-id-2', ...]
+
+    # Boto 2.x
+    ec2_connection.stop_instances(instance_ids=ids)
+    ec2_connection.terminate_instances(instance_ids=ids)
+
+    # Boto 3
+    ec2.instances.filter(InstanceIds=ids).stop()
+    ec2.instances.filter(InstanceIds=ids).terminate()
+
+Checking What Instances Are Running
+-----------------------------------
+Boto 3 collections come in handy when listing all your running instances as well::
+
+    # Boto 2.x
+    reservations = ec2_connection.get_all_reservations(
+        filters={'instance-state-name': 'running'})
+    for reservation in reservations:
+        for instance in reservation.instances:
+            print(instance.instance_id, instance.instance_type)
+
+    # Boto 3
+    instances = ec2.instances.filter(
+        Filters=[{'Name': 'instance-state-name', 'Values': ['running']}])
+    for instance in instances:
```

Comment:
Just wanted to give input to this as a fresh pair of eyes. I actually got tripped up with this. Forgot that ``filter()`` is a method on a collection and not an operation of EC2's API. I just wanted to make a note of it because it took me a few minutes to figure it out.

I think to improve it you could add more descriptive inline comments of what is going on. Like for boto3 it would be:
```
 # Boto 3
# Use the filter() method of the instances collection to retrieve all running ec2 instances.
instances = ec2.instances.filter(
    Filters=[{'Name': 'instance-state-name', 'Values': ['running']}])
```

Would not hurt to add something like that in other similar places in the migration guide. Thoughts?